projects/components/src/filtering/filter/parser/filter-parser-lookup.service.ts

```diff
@@ -1,32 +1,35 @@
 import { Injectable } from '@angular/core';
+import { assertUnreachable } from '@hypertrace/common';
 import { FilterAttributeType } from '../filter-attribute-type';
 import { FilterOperator } from '../filter-operators';
-import { FilterParserConstructor } from './filter-parser-injection-token';
 import { AbstractFilterParser } from './types/abstract-filter-parser';
+import { ComparisonFilterParser } from './types/comparison-filter-parser';
+import { ContainsFilterParser } from './types/contains-filter-parser';
+import { InFilterParser } from './types/in-filter-parser';
 
 @Injectable({
   providedIn: 'root'
 })
 export class FilterParserLookupService {
-  private readonly filterParsers: Map<FilterOperator, AbstractFilterParser<unknown>> = new Map();
-
-  public registerAll(filterParserConstructors: FilterParserConstructor<unknown>[]): void {
-    filterParserConstructors.forEach(filterParserConstructor => {
-      const filterParser = new filterParserConstructor();
-
-      filterParser.supportedOperators().forEach(op => this.filterParsers.set(op, filterParser));
-    });
-  }
-
   public lookup(operator: FilterOperator): AbstractFilterParser<unknown> {
-    const filterParser = this.filterParsers.get(operator);
-
-    if (filterParser === undefined) {
-      throw new Error(`Filter parser not found for operator '${operator}'.
-      Available parsers are '${String(Array.from(this.filterParsers.keys()))}'`);
+    switch (operator) {
+      case FilterOperator.Equals:
+      case FilterOperator.NotEquals:
+      case FilterOperator.LessThan:
+      case FilterOperator.LessThanOrEqualTo:
+      case FilterOperator.GreaterThan:
+      case FilterOperator.GreaterThanOrEqualTo:
+        return new ComparisonFilterParser();
+      case FilterOperator.In:
+        return new InFilterParser();
+      case FilterOperator.ContainsKey:
+      case FilterOperator.ContainsKeyValue:
+        return new ContainsFilterParser();
+      default:
+        assertUnreachable(operator);
     }
 
-    return filterParser;
+    throw new Error(`Filter parser not found for operator '${operator}'.`);
```

Comment:
if you return on line 29 this isn't needed (assertUnreachable returns never because it would throw)